pkg/synthetictests/image_pulls.go

```diff
@@ -0,0 +1,100 @@
+package synthetictests
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	"github.com/openshift/origin/pkg/test/ginkgo/junitapi"
+)
+
+const imagePullRedhatRegEx = `reason/[a-zA-Z]+ .*Back-off pulling image .*registry.redhat.io`
+
+type eventRecognizerFunc func(event monitorapi.EventInterval) bool
+
+func matchEventForRegexOrDie(regex string) eventRecognizerFunc {
+	regExp := regexp.MustCompile(regex)
+	return func(e monitorapi.EventInterval) bool {
+		return regExp.MatchString(e.Message)
+	}
+}
+
+type singleEventCheckRegex struct {
+	testName       string
+	recognizer     eventRecognizerFunc
+	failThreshold  int
+	flakeThreshold int
+}
+
+// test goes through the events, looks for a match using the s.recognizer function,
+// if a match is found, and occurs more than s.failThreshold times, we mark it as a
+// flake (this allows us to see how often this symptom is happening).
+//
+func (s *singleEventCheckRegex) test(events monitorapi.Intervals) []*junitapi.JUnitTestCase {
+	success := &junitapi.JUnitTestCase{Name: s.testName}
+	var failureOutput, flakeOutput []string
+	for _, e := range events {
+		if s.recognizer(e) {
+
+			msg := fmt.Sprintf("%s - %s", e.Locator, e.Message)
+			eventDisplayMessage, times := getTimesAnEventHappened(msg)
+			switch {
+			case times > s.failThreshold:
+				// Threshold exceeded, report it as a failure.
+				//
+				failureOutput = append(failureOutput, fmt.Sprintf("event [%s] happened %d times", eventDisplayMessage, times))
+			case times > 0 && times <= s.flakeThreshold:
```

Comment:
`times > flakeThreshold`

1. times==1, flakeThreshold==5, failThreshold==8 should be  a pass
2. times==6, flakeThreshold==5, failThreshold==8 should be a flake
3. times==9, flakeThreshold==5, failThreshold==8 should be a fail